packages/alfa-dom/test/node/query/element-id-map.spec.tsx

```diff
@@ -0,0 +1,16 @@
+import { test } from "@siteimprove/alfa-test";
+import { h } from "../../../h";
+
+import { Query } from "../../../src";
+
+const { getElementIdMap } = Query;
+
+test("#getElementIdMap() returns map from ids to elements", (t) => {
```

Comment:
**Suggestion:** can we add a test to check that when `id` are duplicated we only return the first one (in tree order)?

Reply:
> We should also use the new cached map for `aria-labelledby` resolution at
> 
> https://github.com/Siteimprove/alfa/blob/b1e42bee45a3cb5d78f50c6af509eeab29519101/packages/alfa-aria/src/name.ts#L890-L903
> 
> (and it should also simplify that code by not needing to sort since we'll poke the map in the correct order)

There's also a follow up task on #1037 on getting rid of all usages of `.elementDescendants()` so it would be picked up there too. Should we keep it as part of that follow up task?